Dockerfile.openshift

```diff
@@ -1,8 +1,6 @@
-# TODO: Switch to UBI golang image when 1.17 is released
-# FROM registry.access.redhat.com/ubi8/go-toolset:latest AS builder
-FROM golang:1.17 AS builder
+FROM registry.access.redhat.com/ubi8/go-toolset:1.16.12 AS builder
 
-WORKDIR /workspace
+WORKDIR /opt/app-root/src
 
 # Copy the dependencies which don't change frequently
```

Comment:
We should reduce layers and just do a `COPY . .`, we can add a `.dockerignore` for other files.

Reply:
That would reduce the layer but increases the build time for the container because the layer changes every time one file in the entire directory changes. Then go has to download all the dependencies, and rebuild everything again.